File: TimeKeeperAndroid/src/com/tjoris/timekeeper/data/IPlaylistStore.java
package com.tjoris.timekeeper.data;

import java.util.List;

public interface IPlaylistStore
{
	public List<PlaylistHeader> readAllPlaylists();
	public void storePlaylistHeader(PlaylistHeader playlist);
	public Playlist readPlaylist(long id);
	public void addPlaylist(Playlist playlist);
	public void deletePlaylist(PlaylistHeader playlist);
	public int getNextPlaylistWeight();
	public void storeSong(Playlist playlist, Song song);
	public void deleteSong(Playlist playlist, Song song);
	public void close();
}

File: TimeKeeperAndroid/app/src/main/java/be/t_ars/timekeeper/data/IPlaylistStore.kt
package be.t_ars.timekeeper.data

interface IPlaylistStore {
    fun readAllPlaylists(): List<PlaylistHeader>
    fun storePlaylistHeader(playlist: PlaylistHeader)
    fun readPlaylist(id: Long): Playlist?
    fun addPlaylist(playlist: Playlist)
    fun deletePlaylist(playlist: PlaylistHeader)
    val nextPlaylistWeight: Int
    fun storeSong(playlist: Playlist, song: Song)
    fun deleteSong(playlist: Playlist, song: Song)
    fun close()
}
